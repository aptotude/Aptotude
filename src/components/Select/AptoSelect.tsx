export { Select as AptoSelect } from 'semantic-ui-react';
import 'semantic-ui-css/semantic.min.css';
