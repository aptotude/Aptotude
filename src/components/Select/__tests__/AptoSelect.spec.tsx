import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import { AptoSelect } from '../AptoSelect';

afterEach(() => {
  cleanup();
});

describe('AptoSelect', () => {
  const options = [
    { key: '0', value: '', text: 'Choose' },
    { key: '1', value: '1', text: 'Item 1' }
  ];

  it('Renders Component', () => {
    const { container } = render(<AptoSelect options={options} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders Component with name/value', () => {
    const { container } = render(
      <AptoSelect options={options} name="hi" value="woo" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Renders disabled', () => {
    const { container } = render(
      <AptoSelect disabled={true} options={options} name="hi" value="woo" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
