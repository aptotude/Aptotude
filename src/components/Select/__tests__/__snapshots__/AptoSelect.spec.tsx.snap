// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoSelect matches snapshot 1`] = `
<DocumentFragment>
  <div
    aria-expanded="false"
    class="ui selection dropdown"
    role="listbox"
    tabindex="0"
  >
    <div
      aria-live="polite"
      class="default text"
      role="alert"
    >
      Select a Person
    </div>
    <i
      aria-hidden="true"
      class="dropdown icon"
    />
    <div
      class="menu transition"
    >
      <div
        aria-checked="false"
        aria-selected="true"
        class="selected item"
        role="option"
        style="pointer-events: all;"
      />
    </div>
  </div>
</DocumentFragment>
`;
